src/wp-includes/wp-db.php

```diff
@@ -2535,6 +2537,20 @@ protected function process_fields( $table, $data, $format ) {
 		$converted_data = $this->strip_invalid_text( $data );
 
 		if ( $data !== $converted_data ) {
+
+			$problem_fields = array();
+			foreach ( $data as $field => $value ) {
+				if ( $value !== $converted_data[ $field ] ) {
+					$problem_fields[] = $field;
+				}
+			}
+
+			$this->last_error = sprintf(
+				/* translators: %s Database field that where the error occurred. */
+				__( 'WordPress database error: Processing the value for the following field failed: %s. The supplied value may be too long or contained invalid data.' ),
```

Comment:
@SergeyBiryukov @ocean90 Is the singular `field` okay if there are more than fields?

Reply:
I think I'd prefer plural `fields` here, but the `1 === count( $problem_fields )` condition from the original patch to differentiate between a singular and plural form would also be OK.

Using `_n()`, on the other hand, would not be correct here, as noted on the ticket:
https://core.trac.wordpress.org/ticket/32315#comment:71

Reply:
Thanks @SergeyBiryukov! Fixed in commit https://github.com/WordPress/wordpress-develop/pull/1874/commits/54f92f2072b94f36914578a0df4ef3561e9bec9a.

Reply:
5 lines of code to handle plurality?

field(s)  

Thanks for finally fixing this but don't overthink the plumbing.